### PR TITLE
Bugfix for container clients failing to retrieve artefacts from S3 storage

### PR DESCRIPTION
**Pull request by m-bull**
Issue described [here](https://pulp.plan.io/issues/9586).

### CODE REVIEW

**Comment by lubosmj** [comment]
@m-bull, would you mind squashing those two commits into a single one and adding a commit message in the format described at https://docs.pulpproject.org/pulpcore/contributing/git.html#commit-message (closes #9586)?

File: CHANGES/9586.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Container clients fail to retrieve artefacts from S3 storage
```
Comment (lubosmj) [inline CHANGES/9586.bugfix:1]: Fixed a bug that caused container clients to be unable to interact with content stored on S3.

File: pulp_container/app/redirects.py [hunk 2]
```diff
@@ -112,7 +112,9 @@ def redirect_to_object_storage(self, artifact, return_media_type):
             "ResponseContentType": return_media_type,
             "ResponseContentDisposition": f"attachment;filename={filename}",
         }
-        content_url = artifact.file.storage.url(artifact.file.name, parameters=parameters)
+        content_url = artifact.file.storage.url(
+            artifact.file.name, parameters=parameters, http_method=self.request.method
+        )
```
Comment (mdellweg) [inline pulp_container/app/redirects.py:117]: Should we add a check, that method is in `rest_framework.permissions.SAFE_METHODS`?

**Comment by ipanova** [comment]
@m-bull I was puzzled on how our tests which also run on S3 pass, and after closely looking at them I found that we actually never push twice same content to the same repo. We need to add a test that would be pushing same image twice, for example here https://github.com/pulp/pulp_container/blob/main/pulp_container/tests/functional/api/test_push_content.py#L81 adding 2 consecutive pushes should cover the bugfix you've submitted.

File: pulp_container/app/redirects.py (hunk 2 above)
Comment (ipanova) [reply pulp_container/app/redirects.py:117]: there is no need to check for safe_methods, redirects are happening anyway only on head/get calls https://github.com/pulp/pulp_container/blob/main/pulp_container/app/registry_api.py#L800 and https://github.com/pulp/pulp_container/blob/main/pulp_container/app/registry_api.py#L832

Comment (ipanova) [reply pulp_container/app/redirects.py:117]: @m-bull we should add this to the core codepath as well https://github.com/pulp/pulpcore/blob/master/pulpcore/content/handler.py#L780

**Comment by ipanova** [comment]
I am merging this and will create a follow-up pr for the mentioned test. Thank you for the contribution!
CI failure are unrelated to the PR changes and should already be fixed on main

**Pull request merged**